### PR TITLE
Don't allow explicit UUIDs and ReplicatedMergeTree arguments in Replicated databases

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Do not allow explicitly specifying UUID when creating a table in Replicated database. Also, do not allow explicitly specifying ZooKeeper path and replica name for *MergeTree tables in Replicated databases

It introduces a new setting `database_replicated_allow_explicit_uuid` and changes the type of `database_replicated_allow_replicated_engine_arguments`. Possible values:
 - 0 - don't allow
 - 1 - allow, log a warning
 - 2 - allow, but ignore user-provided values and use default instead
 - 3 - allow without a warning (for tests)

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [x]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Interpreters/InterpreterCreateQuery.cpp [hunk 1]
```diff
@@ -1055,6 +1055,26 @@ void InterpreterCreateQuery::assertOrSetUUID(ASTCreateQuery & create, const Data
     bool from_path = create.attach_from_path.has_value();
     bool is_on_cluster = getContext()->getClientInfo().query_kind == ClientInfo::QueryKind::SECONDARY_QUERY;
 
+    if (database->getEngineName() == "Replicated" && create.uuid != UUIDHelpers::Nil && !is_replicated_database_internal)
+    {
+        if (getContext()->getSettingsRef().database_replicated_allow_explicit_uuid == 0)
```
Comment (thevar1able) [inline src/Interpreters/InterpreterCreateQuery.cpp:1060]: An enum would be nice.

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 995413662b3784a748bc50abdebf420f400aed26 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66104/995413662b3784a748bc50abdebf420f400aed26/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Interpreters/InterpreterCreateQuery.cpp (hunk 1 above)
Comment (tavplubix) [reply src/Interpreters/InterpreterCreateQuery.cpp:1060]: This setting is not supposed to be usable :)

Comment (thevar1able) [reply src/Interpreters/InterpreterCreateQuery.cpp:1060]: Oh ok nvm.

**Review by thevar1able — APPROVED** [review]

**Pull request merged**